activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -805,10 +805,21 @@ def readonly(value = true)
     end
 
     def readonly!(value = true) # :nodoc:
+      p self.method(:readonly_value).source_location
```

Comment:
✂️

Reply:
Haha oops. This is what I get for opening a PR that I'd been sitting on for months.